### PR TITLE
SYMP-37 | Exit early when user answers "yes" to vulnerable populations questions

### PR DESCRIPTION
**Pull request by taraepp**
Ticket: https://freshworks.atlassian.net/browse/SYMP-37
- Last piece in ensuring that logic in program matches logic in given algorithm (https://docs.google.com/drawings/d/1YrYbu8FUiUz3SU1HwQ9nlSmkQBTdMjsVYwTinqCZnS0)
- exit early with recommendation for no test when no symptoms
- exit early with recommendation for test when yes to any vulnerable groups questions

### CODE REVIEW

File: apps/web/src/forms/SymptomChecker.tsx [hunk 1]
```diff
@@ -65,6 +65,11 @@ export const SymptomChecker = () => {
       return step + 1;
     }
 
+    // exit if no symptoms
+    if (values.symptoms.none.checked === true) {
+      return recommend(Recommendation.ASYMPTOMATIC_NO_TEST);
+    }
```
Comment (arranfw) [inline apps/web/src/forms/SymptomChecker.tsx:71]: AC: When user answers None of the above to the symptoms question (Do you have any of the following symptoms?) preceding the vulnerable groups questions, they should immediately be recommended that they do not need a COVID test.

✅

File: apps/web/cypress/integration/outcome/outcomes.spec.ts [hunk 2]
```diff
@@ -101,7 +98,7 @@ describe('Recommendations', () => {
     cy.contains(questions.q2.question).no().continue();
     cy.contains(questions.q3.question).selectSymptom('fever').selectSymptom('cough').continue();
     cy.contains(questions.severity.cough).get('#Mild').click().continue();
-    cy.contains(questions.hwq1.question).yes().continue();
+    cy.contains(questions.hwq1.question).no().continue();
```
Comment (arranfw) [inline apps/web/cypress/integration/outcome/outcomes.spec.ts:101]: Could you explain this test change?

**Review by arranfw — APPROVED** [review]
AC 1/3 look straightforward, is AC 2 the existing behaviour?

What does removing the length check on `symptoms` do? Is it just an optimization because we can assume they haven't checked any symptoms at this point?

File: apps/web/cypress/integration/outcome/outcomes.spec.ts (hunk 2 above)
Comment (taraepp) [reply apps/web/cypress/integration/outcome/outcomes.spec.ts:101]: Absolutely!
With the changes to their algorithm, selecting "yes" to hwq1 is now supposed to end the assessment with a recommendation to get tested (so the rest of the test code would not be executed). The hwq7/hwq8 responses _also_ trigger a recommendation to get a test, but that logic was trickier so I opted to keep that in.

Comment (arranfw) [reply apps/web/cypress/integration/outcome/outcomes.spec.ts:101]: Ah so previously it didn't matter if it was yes or now, but now yes ends the survey so you've switched it to no?

**Comment by taraepp** [comment]
> AC 1/3 look straightforward, is AC 2 the existing behaviour?
> 
> What does removing the length check on `symptoms` do? Is it just an optimization because we can assume they haven't checked any symptoms at this point?

- AC2 is new behaviour, previously if they answered no to all symptoms they would keep doing the assessment
- Removing length check on symptoms (and the entire symptoms variable) was just optimizing/removing dead code- can assume they have symptoms if they've gotten that far in the assessment and didn't indicate that they have none.

File: apps/web/cypress/integration/outcome/outcomes.spec.ts (hunk 2 above)
Comment (taraepp) [reply apps/web/cypress/integration/outcome/outcomes.spec.ts:101]: Pretty much. Now yes ends the survey so I switched it to no. There were 2 triggers in the test that would give the result.
There's definitely a case to be made for adding more tests to try out more paths.

**Pull request merged**